akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRememberEntitiesSpec.scala

```diff
@@ -130,14 +130,15 @@ abstract class ClusterShardingRememberEntitiesSpec(multiNodeConfig: ClusterShard
 
   lazy val region = ClusterSharding(system).shardRegion(dataType)
 
-  def expectEntityRestarted(
-      sys: ActorSystem,
-      event: Int,
-      probe: TestProbe,
-      entityProbe: TestProbe): EntityActor.Started = {
+  def expectEntityRestarted(sys: ActorSystem, event: Int, entityProbe: TestProbe): EntityActor.Started = {
     if (!multiNodeConfig.rememberEntities) {
-      probe.send(ClusterSharding(sys).shardRegion(dataType), event)
-      probe.expectMsg(1)
+      val probe = TestProbe()(sys)
+      within(20.seconds) {
+        awaitAssert {
+          probe.send(ClusterSharding(sys).shardRegion(dataType), event)
+          probe.expectMsg(1.second, 1)
+        }
+      }
```

Comment:
I compared debug logs and the reason it started failing was that with the quick push `third` already know the location of shard 1 when this message is sent, and then it tries to send it to `second` (which was shutdown but not fully removed yet).

Before the quick push `third` didn't know, and had to GetShardHome. The message was buffered and delivered when GetShardHome was successful (`second` removed).

It would have been the same "message loss" if `third` had some interaction with shard 1 before the shutdown so we haven't introduced any regression with the quick push. (I tried that scenario)

Reply:
Do I get it right if I think that it is also a race around if the region saw the region shard removed, and will fetch it with `GetSharrdHome` before this message arrives or not?

Reply:
That might be the case, but I think the test is using downing so it takes a few seconds for it to be removed.